packages/react/src/components/SearchField/SearchField.test.tsx

```diff
@@ -150,6 +150,31 @@ test('should support name prop', () => {
   expect(input).toHaveDisplayValue('bananas');
 });
 
+test('SearchField should render trailingChildren as a component', async () => {
+  const MyComponent = () => <div className="component">I am a component</div>;
+  render({
+    trailingChildren: <MyComponent />
+  });
+
+  expect(screen.getByText('I am a component')).toHaveClass('component');
+});
+
+test('SearchField should render trailingChildren as a string', async () => {
+  render({
+    trailingChildren: 'I am a component'
+  });
+
+  expect(screen.getByText('I am a component')).toBeInTheDocument();
+});
+
+test('SearchField should render trailingChildren as an element', async () => {
+  render({
+    trailingChildren: <div className="element">I am a component</div>
+  });
+
+  expect(screen.getByText('I am a component')).toHaveClass('element');
+});
+
```

Comment:
Since this is an additional "state" of the component, we should test it with axe below.